### PR TITLE
Stakeholder tool

### PR DESCRIPTION
**Pull request by gwynndp**
DRAFT
- move state from redux to context
- add api functions to mock api
- fix opening the modal for child-orgs

### CODE REVIEW

File: src/components/Stakeholders/Filter.js [hunk 1]
```diff
@@ -0,0 +1,251 @@
...
+  flex: {
+    display: 'flex',
+    flex: 'wrap',
+  },
+  ml: {
+    marginLeft: '24px',
+  },
+});
+
+function Filter() {
```
Comment (nmcharlton) [inline src/components/Stakeholders/Filter.js:38]: Can we call this component `StakeholderFilter`?

File: src/components/Stakeholders/FilterBar.js [hunk 2]
```diff
@@ -0,0 +1,33 @@
...
+  spaceBetween: {
+    justifyContent: 'space-between',
+  },
+  my: {
+    marginTop: '3rem',
+    marginBottom: '2rem',
+  },
+});
+
+export default function FilterBar() {
```
Comment (nmcharlton) [inline src/components/Stakeholders/FilterBar.js:20]: Can this be `StakeholderFilterBar`?

File: src/components/Stakeholders/StakeholderDetail/Header.js [hunk 3]
```diff
@@ -0,0 +1,322 @@
...
+  },
+  inputName: {
+    fontSize: '1.67rem',
+  },
+  input: {
+    height: 31,
+  },
+});
+
+export default function DialogHeader({ data }) {
```
Comment (nmcharlton) [inline src/components/Stakeholders/StakeholderDetail/Header.js:46]: Add _Stakeholder_ to the component name

File: src/components/Stakeholders/Table.js [hunk 4]
```diff
@@ -0,0 +1,95 @@
...
+                    />
+                  ))}
+              </React.Fragment>
+            ))}
+          </TableBody>
+        </Table>
+        <TablePagination
+          component="div"
+          rowsPerPage={rowsPerPage}
+          rowsPerPageOptions={[1, 2, 5]}
```
Comment (nmcharlton) [inline src/components/Stakeholders/Table.js:84]: This doesn't seem right - is it just for testing?

File: src/context/AppContext.js [hunk 5]
```diff
@@ -113,6 +115,18 @@ function getRoutes(user) {
         !user ||
         user.policy.organization !== undefined,
     },
+    {
```
Comment (nmcharlton) [inline src/context/AppContext.js:118]: We have specific policies for stakeholder management now (or at least we will when https://github.com/Greenstand/treetracker-database/pull/92 is merged): `manage_stakeholders` and `list_stakeholders`

File: src/models/FilterStakeholder.js [hunk 6]
```diff
@@ -0,0 +1,148 @@
+// import { stringToSearchRegExp } from '../utilities';
+
+/*
+ * A simple model for grower filter
+ */
+
+export const ALL_ORGANIZATIONS = 'ALL_ORGANIZATIONS';
+export const ORGANIZATION_NOT_SET = 'ORGANIZATION_NOT_SET';
+
+export default class Filter {
```
Comment (nmcharlton) [inline src/models/FilterStakeholder.js:10]: Should be `FilterStakeholder`

File: src/models/FilterStakeholder.js [hunk 7]
```diff
@@ -0,0 +1,148 @@
...
+  constructor(options) {
+    Object.assign(this, options);
+  }
+
+  getWhereObj() {
```
Comment (nmcharlton) [inline src/models/FilterStakeholder.js:15]: Are we using LoopBack query syntax with the new API?

File: src/models/FilterStakeholder.js [hunk 8]
```diff
@@ -0,0 +1,148 @@
...
+        }
+
+        if (this.lastName) {
+          where.lastName = lastName[0];
+        }
+        where.orgName = '';
+      } else {
+        // give all name fields the value for filter
+        where.orgName = this.name;
+        where.firstName = firstName || this.name;
```
Comment (nmcharlton) [inline src/models/FilterStakeholder.js:36]: Setting the `firstName` and `lastName` in the filter for a non-person doesn't seem right to me.

Comment (nmcharlton) [reply src/models/FilterStakeholder.js:36]: I wonder if the logic for constructing a name filter is the sort of more sophisticated task we should delegate to the API.

**Review by nmcharlton — COMMENTED** [review]
Great work @gwynndp, this is looking great.

My main comment is to ensure we include the module name (_Stakeholder(s)_) in the names of components that are specific to that module. I understand omitting it from the filename for brevity, but it should be in the component name.

File: src/models/FilterStakeholder.js (hunk 7 above)
Comment (gwynndp) [reply src/models/FilterStakeholder.js:15]: I wasn't sure if we should keep it uniform from the client, so I did, but we don't have to.  There is validation for what can be filtered on the API as well.

File: src/components/Stakeholders/Table.js (hunk 4 above)
Comment (gwynndp) [reply src/components/Stakeholders/Table.js:84]: It is the count of the main (logged in) stakeholder rows, not including their children, but you're right I'll revisit

File: src/context/AppContext.js (hunk 5 above)
Comment (gwynndp) [reply src/context/AppContext.js:118]: I added those policies with the current implementation but I think we'll have to revisit after the new policies are merged.

**Pull request closed without merging**